docs/dev/new_check_howto.md

```diff
@@ -88,187 +91,166 @@ A Check is a Python class with the following requirements:
 * It must derive from `AgentCheck`
 * It must provide a method with this signature: `check(self, instance)`
 
-Checks are organized in regular Python packages under the `datadog_checks` namespace, so your code should live under `my_check/datadog_checks/my_check`. The only requirement is that the name of the package has to be the same as the check name. There are no particular restrictions on the name of the Python modules within that package, nor on the name of the class implementing the check.
+Checks are organized in regular Python packages under the `datadog_checks` namespace, so your code should live under `awesome/datadog_checks/awesome`. The only requirement is that the name of the package has to be the same as the check name. There are no particular restrictions on the name of the Python modules within that package, nor on the name of the class implementing the check.
 
 ### Implement check logic
 
-Let's say we want to create a Service Check named `my_check` that checks for a string on a web page. It will result in `OK` if the string is present, `WARNING` if the page is accessible but the string was not found, and `CRITICAL` if the page is inaccessible.
+Let's say we want to create a Service Check named `awesome.search` that searches for a string on a web page. It will result in `OK` if the string is present, `WARNING` if the page is accessible but the string was not found, and `CRITICAL` if the page is inaccessible.
 
 The code would look something like this:
 
 ```python
 import requests
 
-from datadog_checks.checks import AgentCheck
-from datadog_checks.errors import CheckException
+from datadog_checks.base import AgentCheck, ConfigurationError
 
 
-# MyCheck derives from AgentCheck, and provides the required check method.
-class MyCheck(AgentCheck):
+class AwesomeCheck(AgentCheck):
+    """AwesomeCheck derives from AgentCheck, and provides the required check method."""
+
     def check(self, instance):
         url = instance.get('url')
         search_string = instance.get('search_string')
 
-        # It's a good idea to do some basic sanity checking. Try to be as
-        # specific as possible, with the exceptions; you can fall back to
-        # CheckException when in doubt though.
+        # It's a very good idea to do some basic sanity checking.
+        # Try to be as specific as possible with the exceptions.
         if not url or not search_string:
-            raise CheckException("Configuration error, please fix my_check.yaml")
+            raise ConfigurationError('Configuration error, please fix awesome.yaml')
```

Comment:
Style question: should we recommend single or double quotes for output strings?

Reply:
Single is what we usually do